src/pc/djui/djui_panel_controls.c

```diff
@@ -28,18 +28,53 @@ void djui_panel_controls_create(struct DjuiBase* caller) {
         int numJoys = SDL_NumJoysticks();
         if (numJoys == 0) { numJoys = 1; }
         if (numJoys > 10) { numJoys = 10; }
-        int strSize = numJoys * 2;
+
         char* gamepadChoices[numJoys];
-        char gamepadChoicesLong[strSize];
+
+        // Get the names of all connected gamepads, if none is provided, use "Unknown"
         for (int i = 0; i < numJoys; i++) {
-            int index = i * 2;
-            if (i > 9) {
-                index += (i - 9);
+            const char* joystickName = SDL_JoystickNameForIndex(i);
+            if (joystickName == NULL) {
+                joystickName = "Unknown";
             }
-            sprintf(&gamepadChoicesLong[index], "%d", i);
-            gamepadChoices[i] = &gamepadChoicesLong[index];
+            gamepadChoices[i] = strdup(joystickName);
         }
+
+        // Check for repeated names and append a number if necessary
+        for (int i = 0; i < numJoys; i++) {
+            int count = 1;
+            for (int j = 0; j < i; j++) {
+                if (strcmp(gamepadChoices[i], gamepadChoices[j]) == 0) {
+                    count++;
+                    char newName[256];
+                    
+                    // If the name is bigger than 9 characters, we need to truncate it first
+                    // Then we can append the number so it fits in the slot
+
+                    // Should we look into making scrolling text for this?
+                    if (strlen(gamepadChoices[i]) > 9) {
+                        strncpy(newName, gamepadChoices[i], 9);
+                        newName[9] = '\0';
+                        strcat(newName, "...");
+                        snprintf(newName, sizeof(newName), "%s (%d)", newName, count);
```

Comment:
using `newName` in `snprintf` as both the destination and an argument is undefined behavior in C
but you don't need to do that, printf functions have a specific format for trimming:
```
snprintf(newName, sizeof(newName), "%.9s... (%d)", gamepadChoices[i], count);
```

Reply:
Alright, changed it, hopefully I did it right heh

Reply:
keep the if statement, otherwise you'll have the ellipsis even for names that have less than 9 characters

Reply:
Ah, right my bad